paimon-core/src/main/java/org/apache/paimon/utils/BranchManager.java

```diff
@@ -133,4 +141,43 @@ public boolean branchExists(String branchName) {
         Path branchPath = branchPath(branchName);
         return fileExists(branchPath);
     }
+
+    /** Get branch count for the table. */
+    public long branchCount() {
+        try {
+            return listVersionedFileStatus(fileIO, branchDirectory(), BRANCH_PREFIX).count();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /** Get all branches for the table. */
+    public List<TableBranch> branches() {
+        try {
+            List<Pair<Path, Long>> paths =
+                    listVersionedFileStatus(fileIO, branchDirectory(), BRANCH_PREFIX)
+                            .map(status -> Pair.of(status.getPath(), status.getModificationTime()))
+                            .collect(Collectors.toList());
+            List<TableBranch> branches = new ArrayList<>();
+            for (Pair<Path, Long> path : paths) {
+                String branchName = path.getLeft().getName().substring(BRANCH_PREFIX.length());
+                FileStoreTable branchTable =
+                        FileStoreTableFactory.create(
+                                fileIO, new Path(getBranchPath(tablePath, branchName)));
+                Snapshot snapshot = branchTable.snapshotManager().latestSnapshot();
```

Comment:
`branchTable.snapshotManager().latestSnapshot()` will return lastest snapshot in main branch.